src/iceberg/table_scan.cc

```diff
@@ -45,6 +43,21 @@ int32_t FileScanTask::files_count() const { return 1; }
 
 int64_t FileScanTask::estimated_row_count() const { return data_file_->record_count; }
 
+Result<std::unique_ptr<ArrowArrayReader>> FileScanTask::ToArrowArrayReader(
```

Comment:
Shouldn't a `FileScanTask` return a stream of `RecordBatch`? Not `ArrowArray`?

Reply:
We have two libraries, one is `libiceberg` and the other is `libiceberg-bundle`. For `libiceberg`, it only depends on Arrow C ABI so it cannot have access to `::arrow::RecordBatch`. An alternative is to use `ArrowArrayStream` but it requires more work so I'm not sure if it worths the effort at this moment. Since the API is not stable before 1.0.0, we can change our mind at any time for any better idea.